src/main.rs

```diff
@@ -0,0 +1,447 @@
+    fs::create_dir_all(ldk_data_dir.clone()).unwrap();
+
+    // Initialize our bitcoind client.
+    let bitcoind_client = match BitcoindClient::new(args.bitcoind_rpc_host.clone(),
+                                         args.bitcoind_rpc_port, args.bitcoind_rpc_username.clone(),
+                                         args.bitcoind_rpc_password.clone()) {
+        Ok(client) => Arc::new(client),
+        Err(e) => {
+            println!("Failed to connect to bitcoind client: {}", e);
+            return
+        }
+    };
+    let mut bitcoind_rpc_client = bitcoind_client.get_new_rpc_client().unwrap();
+
+    // ## Setup
+    // Step 1: Initialize the FeeEstimator
+
+    // BitcoindClient implements the FeeEstimator trait, so it'll act as our fee estimator.
+    let fee_estimator = bitcoind_client.clone();
+
+    // Step 2: Initialize the Logger
+    let logger = Arc::new(FilesystemLogger::new(ldk_data_dir.clone()));
+
+    // Step 3: Initialize the BroadcasterInterface
+
+    // BitcoindClient implements the BroadcasterInterface trait, so it'll act as our transaction
+    // broadcaster.
+    let broadcaster = bitcoind_client.clone();
+
+    // Step 4: Initialize Persist
+    let persister = Arc::new(FilesystemPersister::new(ldk_data_dir.clone()));
+
+    // Step 5: Initialize the ChainMonitor
+    let chain_monitor: Arc<ArcChainMonitor> = Arc::new(ChainMonitor::new(None, broadcaster.clone(),
+                                                           logger.clone(), fee_estimator.clone(),
+                                                           persister.clone()));
+
+    // Step 6: Initialize the KeysManager
+
+    // The key seed that we use to derive the node privkey (that corresponds to the node pubkey) and
+    // other secret key material.
+    let keys_seed_path = format!("{}/keys_seed", ldk_data_dir.clone());
+	  let keys_seed = if let Ok(seed) = fs::read(keys_seed_path.clone()) {
+		    assert_eq!(seed.len(), 32);
+		    let mut key = [0; 32];
+		    key.copy_from_slice(&seed);
+		    key
+	  } else {
+		    let mut key = [0; 32];
+		    thread_rng().fill_bytes(&mut key);
+		    let mut f = File::create(keys_seed_path).unwrap();
+		    f.write_all(&key).expect("Failed to write node keys seed to disk");
+		    f.sync_all().expect("Failed to sync node keys seed to disk");
+		    key
+	  };
+	  let cur = SystemTime::now().duration_since(SystemTime::UNIX_EPOCH).unwrap();
+    let keys_manager = Arc::new(KeysManager::new(&keys_seed, cur.as_secs(), cur.subsec_nanos()));
+
+    // Step 7: Read ChannelMonitor state from disk
+    let monitors_path = format!("{}/monitors", ldk_data_dir.clone());
+    let mut outpoint_to_channelmonitor = disk::read_channelmonitors(monitors_path.to_string(),
+                                                                    keys_manager.clone()).unwrap();
+
+    // Step 9: Initialize the ChannelManager
+    let user_config = UserConfig::default();
+    let runtime = Runtime::new().unwrap();
```

Comment:
It kinda sucks to have multiple runtimes (here and bitcoind_client, at least). Instead, does it make sense to just add `#[tokio::main]` to `main()` and never create runtimes at all?

Reply:
I find it less intimidating to not have `#[tokio::main]` and `async`s everywhere, feel like the glue code should get out of LDK's way as much as possible to make the LDK core concepts clear. But maybe Steve or someone less familiar can give a take on what would seem less intimidating for beginners. Would be open to having a version of the sample that's more "production" that doesn't use the extra runtime.

Reply:
Hmm, I find it relatively intuitive, especially if you can avoid having a bunch of calls to some opaque `runtime` object, and certainly in the case like `main()` where you never actually block on any of the futures, it allows you to just call `tokio::spawn()` instead of `runtime.spawn()`, which I think is more readable.

In the contexts where we want to call async functions from a sync context, you can always call tokio::spawn and forget it, though of course that means we get no error handling. Sadly tokio *really* doesn't support calling async methods from sync contexts, which is kinda dumb, but that's how it is. That said, its...somewhat awkward that our sample client is somewhat gratuitously inefficient - multiple thread pools that are each mostly empty, etc.

Reply:
I'll think on this more, but I ran into issues with `tokio::spawn`  never running my functions unless it was tagged with an `await` at the end (thus making it impossible to run `async` functions in `sync` contexts). Whereas, `runtime.block_on` *does* allow me to run async functions from sync contexts.

Reply:
> I'll think on this more, but I ran into issues with tokio::spawn never running my functions unless it was tagged with an await at the end

Hmm, as long as the last thing the spawn closure does is await or returns the future, it should work (ie `tokio::spawn(|| { X.await })` or `tokio:spawn(|| { X })`), at least as long as the runtime in question continues to exist (ie via `tokio::main`). 

> Eh, not for a tutorial IMO stuck_out_tongue

Sure, but its also a demo, we should, where its easy, we should make at least marginal effort to ensure the code demonstrates not just how easy it is to build a node with LDK, but how easy it is to build an efficient and lightweight node with LDK. One of our selling points is low memory usage, if we spawn 20 threads we definitely won't showcase that :).

Reply:
Hm yeah, I see what you mean. FYI started switching away from multiple runtimes!